examples/post-processing/rs-post-processing.cpp

```diff
@@ -0,0 +1,411 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include <librealsense2/rs.hpp> // Include RealSense Cross Platform API
+#include "example.hpp"          // Include short list of convenience functions for rendering
+
+#include <imgui.h>
+#include "imgui_impl_glfw.h"
+
+// a time interval which must pass between movement of the pointcloud
+#define ROTATION_DELTA 1e7
+
+// maximum angle for the rotation of the pointcloud
+#define MAX_ANGLE 90
+
+// for the interface: distance between checkbox and slider
+#define DIST 120
+
+
+// Struct for managing the pointcloud view
+struct state {
+    state() : yaw(0.0), pitch(0.0), offset_x(0.0f), offset_y(0.0f), tex() {}
+    double yaw, pitch; float offset_x, offset_y; texture tex;
+};
+
+// Struct of parameters controled by the user
+struct params {
+    bool do_decimate, do_spatial, do_temporal, do_disparity;
+    int decimate_magnitude, temporal_magnitude, spatial_magnitude, temp_smooth_delta, spat_smooth_delta;
+    float temp_smooth_alpha, spat_smooth_alpha;
+};
+
+struct filter_ranges {
+    rs2::option_range dec_magnitude_range, spat_magnitude_range, spat_smooth_alpha_range,
```

Comment:
Perhaps some more readable data-structure can be used, like map from option to range?